shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1799,11 +1857,74 @@ - (void)scrollEvent:(UIPanGestureRecognizer*)recognizer API_AVAILABLE(ios(13.4))
     _mouseState.last_translation = CGPointZero;
   }
 
+  if (_mouseState.flutter_state_is_added) {
+    packet->SetPointerData(/*index=*/0, pointer_data);
+  } else {
+    flutter::PointerData add_pointer_data = pointer_data;
+    add_pointer_data.signal_kind = flutter::PointerData::SignalKind::kNone;
+    add_pointer_data.change = flutter::PointerData::Change::kAdd;
+    _mouseState.flutter_state_is_added = true;
+    packet->SetPointerData(/*index=*/0, add_pointer_data);
+    packet->SetPointerData(/*index=*/1, pointer_data);
+  }
+
+  [_engine.get() dispatchPointerDataPacket:std::move(packet)];
+}
+
+- (void)panEvent:(UIPanGestureRecognizer*)recognizer API_AVAILABLE(ios(13.4)) {
+  CGPoint translation = [recognizer translationInView:self.view];
+  const CGFloat scale = [UIScreen mainScreen].scale;
+
+  translation.x *= scale;
+  translation.y *= scale;
+
+  flutter::PointerData pointer_data = [self generatePointerDataForMouse];
+  pointer_data.kind = flutter::PointerData::DeviceKind::kTrackpad;
+  pointer_data.device = reinterpret_cast<int64_t>(recognizer);
+  if (recognizer.state == UIGestureRecognizerStateBegan) {
+    pointer_data.change = flutter::PointerData::Change::kPanZoomStart;
+  } else if (recognizer.state == UIGestureRecognizerStateChanged) {
+    pointer_data.change = flutter::PointerData::Change::kPanZoomUpdate;
+    pointer_data.pan_x = translation.x;
+    pointer_data.pan_y = translation.y;
+    pointer_data.pan_delta_x = 0;  // Delta will be generated in pointer_data_packet_converter.cc.
+    pointer_data.pan_delta_y = 0;  // Delta will be generated in pointer_data_packet_converter.cc.
+    pointer_data.scale = 1;
+  } else {
+    pointer_data.change = flutter::PointerData::Change::kPanZoomEnd;
```

Comment:
For trackpad gesture we will only have kPanZoomStart, kPanZoomUpdate, kPanZoomEnd. I didn't make a separate PanZoomCancelled since I didn't find any situation where it would be used differently than PanZoomEnd in the framework.

Reply:
we should probably still use cancel just to be safe, also can this be a switch statements and put assert on the case that are unexpected?

Reply:
I updated to use switch so it's more obvious what UIKit phase maps to which flutter phase.